### ISSUE TITLE
Deleted half my files (Even local backups)

### ISSUE THREAD

**Issue by Excadite** [stravo1/obsidian-gdrive-sync#27]
When downloading files it errors and deleted half of the files.

I dont have the logs because it deleted them too

DO NOT USE THIS PLUGIN ITS UNSTABLE

**Comment by stravo1** [comment]
also can you please describe what you mean by "local backups"? the plug-in (in fact any plug-in) has access to only the files in the vault, if you created a back-up of the vault in some other location, it cannot access those backups.

did you mean the local files on your vault?

**Comment by jack5github** [comment]
I experienced the exact same problem as the original poster. Fortunately I am a new user of Obsidian, so this wasn't as big of an issue as it may have been. As I am a new user, I have been attempting to create and import files from various sources. I found myself annoyed by this plugin because it was arbitrarily deciding what to sync, was never privy to when I renamed files, and failed to start syncing files when I clicked the Upload File and Upload All Files buttons. Frustrated with the plugin, I deleted all files from the Google Drive backup and cleared the "filesList" key in data.json. To my horror, all of the notes that I had created were deleted the next time I opened the application, with absolutely no way of getting them back. Deleting files like this without confirmation is an absolutely terrible way to handle syncing in beta. Please fix the project so that it simply moves files to a `obsidian-gdrive-sync-deleted` folder instead of removing the files themselves.

**Comment by stravo1** [comment]
hi,

thank you for providing an alternative solution to the deletion issue that exists, i think this will greatly help in recovering mistakenly deleted files.

however deleting files from Google Drive without deleting the vault folder signals to the plug-in that some other device might have deleted them from the Drive, this is because the files which were once synced (had the lastSync tag), are no more there in the Drive, which tells the plug-in that another device has deleted those files and thus in order to keep the vault in sync the files were deleted from your local vault as well.

the Upload All button will be removed in future updates because it's no longer required as notes will auto upload even if they were somehow not uploaded initially.

renaming files in vault renames them on Drive but at times it might happen that the original file (file before renaming) is downloaded again and then deleted or the renamed file gets deleted and then downloaded again, this happens in some race conditions. These situations should not arise frequently but even in such cases no data should be lost.

**Comment by stravo1** [comment]
files are now moved to ".trash" folder under the vault instead of being deleted permanently at https://github.com/stravo1/obsidian-gdrive-sync/releases/tag/v0.9.9-beta-11

**Comment by stravo1** [comment]
hi,
unfortunately the plugin does not track versions, new content overwrites the old content, although there is a plan to incorporate at-least one backup file for each note (i.e., just the version before it is overwritten) in a future update.

although the following suggestion does not guarantee a solution to your issue but you can follow it to get less such issues:
always use the plugin with good internet, and after making a change in a note wait till the lastSync tag/property changes to ensure the change was uploaded to Drive before closing the note/app, in desktop there is a progress indicator as well in the lower right corner to show the syncing status.

thanks.

**Comment by ParthoM7** [comment]
> hi, unfortunately the plugin does not track versions, new content overwrites the old content, although there is a plan to incorporate at-least one backup file for each note (i.e., just the version before it is overwritten) in a future update.
> 
> although the following suggestion does not guarantee a solution to your issue but you can follow it to get less of such issues: always use the plugin with good internet, and after making a change in a note wait till the lastSync tag/property changes to ensure the change was uploaded to Drive before closing the note/app, in desktop there is a progress indicator as well in the lower right corner to show the syncing status.
> 
> thanks.

Sure, thanks. Should it be possible to somehow recover the versions from the Google Drive trash?

**Comment by stravo1** [comment]
no, the files are overwritten, so no versioning happens at all currently.
